.eslintignore

```diff
@@ -0,0 +1,3 @@
+.eslintrc.js
```

Comment:
I don't get any linting errors, but my VS Code formatter does highlight this as error, which is kind of annoying. My default formatter is Prettier. any idea how I can resolve this?

Reply:
Intereting, I haven't checked the vscode formatter, but indeed I see the error now.

Ok, I've investigated a bit more and it looks like all the .js files are not working with linter anymore and not only `eslintrc.js`. (for example the same error exists in `webpack.config.js` and other js files. They are not caught with the`lint` script because it's only running the eslint inside `src` directory: `eslint src/**/*.{js,ts,tsx}`.

So we should figure out a generic way to fix this instead of adding them to the gitignore file.
I've figured out a working solution but it involves a few changes:

```diff
diff --git a/.eslintrc.js b/.eslintrc.js
index ac7ebf0..fb9bbd6 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -1,21 +1,20 @@
 module.exports = {
   root: true,
+  "env": {
+    "browser": true,
+    "node": true
+  },
   parser: '@typescript-eslint/parser',
   parserOptions: {
-    tsconfigRootDir: __dirname,
-    project: ['./tsconfig.json'],
     ecmaFeatures: {
       jsx: true,
     },
   },
   plugins: ['@typescript-eslint', 'import', 'jest', 'react'],
   extends: [
-    'plugin:@typescript-eslint/recommended',
-    'plugin:@typescript-eslint/recommended-requiring-type-checking',
     'plugin:react/recommended',
     'plugin:react/jsx-runtime',
     'plugin:jest/recommended',
-    'airbnb-typescript',
     'prettier',
   ],
   settings: {
@@ -52,6 +51,24 @@ module.exports = {
     'react/jsx-uses-react': 'error',
   },
   overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
+      // We should extend TypeScript plugins here instead of extending them
+      // outside the `overrides`.
+      extends: [
+        'plugin:@typescript-eslint/recommended',
+        'plugin:@typescript-eslint/recommended-requiring-type-checking',
+        'airbnb-typescript',
+      ],
+      parserOptions: {
+        tsconfigRootDir: __dirname,
+        project: ['./tsconfig.json'],
+      },
+      rules: {
+        // https://github.com/typescript-eslint/typescript-eslint/issues/1824
+        '@typescript-eslint/indent': 'off',
+      },
+    },
     {
       // typescript test files
       files: ['src/tests/**/*.{ts,tsx}'],
```
These changes are what makes it work for me with the js files as well. There are a few things to unpack though:
1. We need to add `env` object here because we need to tell eslint that this code runs on both web **and** node. The "node" field here is critical because otherwise `module.exports` in this file will give error thinking that there is no `modules` global variable (which only exists on node).
2. We need to create another override here for ts and tsx files unfortunatly. I know this makes the code a bit more crowded, but this is actually needed to isolate ts and tsx file specific rules. We are also moving the typescript related `extends` here.
3. Most importantly `project: ['./tsconfig.json'],` field should be inside the override as well. This is telling linter that only typescript files are subject to tsconfig.json rules. Otherwise eslint will try to assume other .js files are also typescript.
4. Lastly we have `'@typescript-eslint/indent': 'off',` rule here because of a known eslint bug :( This was there before actually. We didn't need it in the new version because we didn't have .ts specific rules, but that problem is back again.